hudi-common/src/main/java/org/apache/hudi/common/util/SpillableMapUtils.java

```diff
@@ -152,10 +153,13 @@ public static <R> HoodieRecord<R> convertToHoodieRecordPayload(GenericRecord rec
       record = recordWithoutMetaFields;
     }
 
-    HoodieRecord<? extends HoodieRecordPayload> hoodieRecord = new HoodieAvroRecord<>(new HoodieKey(recKey, partitionPath),
-        HoodieRecordUtils.loadPayload(payloadClazz, record, preCombineVal), operation);
-
-    return (HoodieRecord<R>) hoodieRecord;
+    if (StringUtils.isNullOrEmpty(payloadClazz)) {
+      return (HoodieRecord<R>) new HoodieAvroIndexedRecord(new HoodieKey(recKey, partitionPath), record);
```

Comment:
we can not use this, record would contain a schema schema field which is huge.

Reply:
I am going to revert this change to unblock landing this patch. but we might need to fix this as part of fixing HoodieRecordCreation. https://issues.apache.org/jira/browse/HUDI-9713